org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/findReferences/WorkspaceResourceAccess.xtend

```diff
@@ -25,7 +25,11 @@ class WorkspaceResourceAccess implements IResourceAccess {
 
 	override <R> readOnly(URI targetURI, IUnitOfWork<R, ResourceSet> work) {
 		return workspaceManager.doRead(targetURI) [ document, resource |
-			return work.exec(resource.resourceSet)
+			val resourceSet = resource?.resourceSet
+			if (resourceSet === null) {
+				return null
```

Comment:
Also what you really seem to check here is `resource === null`. Checking for the resourceSet was a bit missleading to me. Are you really considering a case, where there is a resource but not resourceSet?